BrainPortal/app/models/tool_config.rb

```diff
@@ -397,43 +416,75 @@ def validate_container_rules #:nodoc:
     return errors.empty?
   end
 
-  # Verify that the admin has entered a set of
+  # breaks down overlay spec onto a list of overlays
+  def parsed_overlay_specs(specs)
```

Comment:
Make this a method that doesn't take any arguments. It should itself fetch the value of the raw specs with `self.singularity_overlays_specs.presence` and return the parsed version.

That means everywhere else you won't need to extract the specs and pass it in argument. It will make the code simpler at the beginning of `singularity_overlays_full_paths` and at the begining of `data_providers_with_overlays`

Reply:
ok, btw `presence`   seems to be redundant, as `blank?` is used in the following statement.